src/Transport/AzureServiceBusSettings.cs

```diff
@@ -0,0 +1,179 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using Microsoft.Azure.ServiceBus;
+    using Microsoft.Azure.ServiceBus.Primitives;
+
+    /// <summary>
+    /// Provides a backwards-compatible configuration API for <see cref="AzureServiceBusTransport"/>.
+    /// </summary>
+    public class AzureServiceBusSettings : TransportSettings<AzureServiceBusTransport>
+    {
+        internal AzureServiceBusSettings(AzureServiceBusTransport transport, RoutingSettings<AzureServiceBusTransport> routing) : base(transport, routing)
+        {
+        }
+
+        /// <summary>
+        /// Specifies a callback to apply to the subscription name when the endpoint's name is longer than 50 characters.
+        /// </summary>
+        [ObsoleteEx(
+            ReplacementTypeOrMember = "SubscriptionNamingConvention",
```

Comment:
Should this one direct to `AzureServiceBusTransport.SubscriptionNamingConvention` rather than to another obsoleted method?
```suggestion
            ReplacementTypeOrMember = "AzureServiceBusTransport.SubscriptionNamingConvention",
```

Reply:
updated